### DISCUSSION TITLE
Boundary conditions and heat equation

### DISCUSSION THREAD

**Discussion by SebfraOslo** [kinnala/scikit-fem#933] [Q&A] [answered]
Hi,

I am trying to impose Dirichlet boundary conditions (temperature) on a heat transfer equation problem (similar to example 19 in scikit-fem documentation). 
I use the function "penalize" (by the way, is it a scikit-fem function?) to do this:

bc = basis.zeros()
bc[basis.get_dofs('bottom')] = 1000
bc[basis.get_dofs('top')] = 0
L0, M0 = penalize(L, M, x=bc, D=basis.get_dofs({'bottom', 'top'}))

It recognizes well the bottom and top boundaries but always put 0 on the two boundaries (instead of 1000 at the bottom). I suppose I am doing something wrong but what? 

Thank you
Sebastien

**Comment by kinnala** [comment, marked as answer]
I think this is not how you set inhomogeneous boundary condition in a time-dependent problem.

I modified the evolution loop as follows in ex19.py:
```python
    while np.linalg.norm(u, np.inf) > 2**-3:
        t = t + dt
        u[basis.get_dofs(lambda x: np.isclose(x[1], 3.0))] = 1
        u = solve(*penalize(A.T, B @ u, x=u, D=basis.get_dofs()))
        yield t, u
```
In other words, you need to enforce the boundary condition at every step over and over again.

**Reply by SebfraOslo** [reply]
Thank you for your answer. It works nicely now

Sebastien
